lib/pub_sub/poller.rb

```diff
@@ -1,31 +1,38 @@
 module PubSub
   class Poller
-    def initialize(queue_url, verbose = false)
-      @queue_url = queue_url
+    def initialize(verbose = false)
       @verbose = verbose
     end
 
     def poll
-      poller.poll(config) do |message|
-        if @verbose
-          PubSub.logger.info(
-            "PubSub received: #{message.message_id} - #{message.body}"
-          )
+      loop do
+        Breaker.current_breaker.run do
+          poller.poll(idle_timeout: 60) do |message|
+            if @verbose
+              PubSub.logger.info(
+                "PubSub received: #{message.message_id} - #{message.body}"
+              )
+            end
+            Message.new(message.body).process
+          end
         end
-        Message.new(message.body).process
+        Breaker.use_next_breaker
```

Comment:
Nice! The replicated try/catching was bugging me but I couldn't quite think of a better option. Will do.